### DISCUSSION TITLE
Regarding the yaml file of the yolov8 series

### DISCUSSION THREAD

**Discussion by wxgkuaile** [ultralytics/ultralytics#14649] [Q&A]
Regarding the yaml file of the yolov8 series, is it generated by itself?
I wrote a python file like this and ran it. Is the yolov8s.yaml here generated by itself?
```python
from ultralytics import YOLO

# Load a model
model = YOLO("yolov8s.yaml")  # build a new model from scratch
model = YOLO("yolov8s.pt")  # load a pretrained model (recommended for training)

# Train the model
results = model.train(data="allanddemostration.yaml", epochs=40, imgsz=640, batch=32, optimizer='SGD')
# Train the model with 2 GPUs
#results = model.train(data='coco128.yaml', epochs=100, imgsz=640, device=[0,1])

# Evaluate the model's performance on the validation set
#results = model.val()

# Export the model to ONNX format
success = model.export(format="onnx")
```

**Comment by pderrenger** [comment]
@wxgkuaile the `yolov8s.yaml` file is not generated automatically. It is a configuration file that defines the architecture and hyperparameters of the YOLOv8s model. You need to have this file available in your working directory or specify its path when building a new model from scratch. If you don't have this file, you can find the default configuration files in the Ultralytics GitHub repository or create your own based on the required model specifications. For more details, you can refer to the [Ultralytics YOLOv8 documentation](https://docs.ultralytics.com/models/yolov8/).
